src/createBaseRouter.js

```diff
@@ -6,9 +6,24 @@ import warning from 'warning';
 
 import { routerShape } from './PropTypes';
 import RedirectException from './RedirectException';
+import createRender from './createRender';
 import resolveRenderArgs from './utils/resolveRenderArgs';
 
-export default function createBaseRouter({ render }) {
+export default function createBaseRouter({
+  render,
+  renderPending,
+  renderReady,
+  renderError,
```

Comment:
practically i've never seen us save bytes here before. you almost always end up using createRender